olmo/train.py

```diff
@@ -131,7 +136,59 @@ def cross_entropy_loss(
     elif reduction == "sum":
         z_squared = (z_squared * (labels != ignore_index)).sum()
 
-    z_loss = 1e-4 * z_squared
+    z_loss = z_loss_multiplier * z_squared
+
+    return loss, z_loss
+
+
+def fused_loss_fn(
+    logits,
+    labels,
+    ignore_index: int = -100,
+    reduction: str = "mean",
+    compute_z_loss: bool = False,
+    z_loss_multiplier: float = 1e-4,
+):
+    import flash_attn
+    from flash_attn.ops.triton.cross_entropy import cross_entropy_loss  # type: ignore
```

Comment:
I worry about importing modules within a function that's called repetitively. Alternatively you could throw this whole definition within a `try import flash_attn: ... except ImportError` block. That would probably be more efficient at runtime.